dojo/tools/pip_audit/parser.py

```diff
@@ -1,70 +1,115 @@
-                        severity="Medium",
-                        description=description,
-                        mitigation=mitigation,
-                        component_name=component_name,
-                        component_version=component_version,
-                        vuln_id_from_tool=vuln_id,
-                        static_finding=True,
-                        dynamic_finding=False,
-                    )
-                    vulnerability_ids = list()
-                    if vuln_id:
-                        vulnerability_ids.append(vuln_id)
-                    if vulnerability_ids:
-                        finding.unsaved_vulnerability_ids = vulnerability_ids
-
-                    findings.append(finding)
+        findings = None
+        # this parser can handle two distinct formats see sample scan files
+        if "dependencies" in data:
+            # new format of report
+            findings = get_file_findings(data, test)
+        else:
+            # legacy format of report
+            findings = get_legacy_findings(data, test)
 
         return findings
+
+
+def get_file_findings(data, test):
+    """Return the findings in the vluns array inside the dependencies key."""
+    findings = list()
+    for dependency in data["dependencies"]:
+        logger.debug("**-**")
+        logger.debug(dependency)
+        item_findings = get_item_findings(dependency, test)
+        if item_findings is not None:
+            findings.extend(item_findings)
+    return findings
+
+
+def get_legacy_findings(data, test):
+    """Return the findings gathered from the vulns element."""
+    findings = list()
+    for item in data:
+        item_findings = get_item_findings(item, test)
+        if item_findings is not None:
+            findings.extend(item_findings)
+    return findings
+
+
+def get_item_findings(item, test):
+    """Return list of Findings."""
+    findings = list()
+    vulnerabilities = item.get("vulns", [])
+    if vulnerabilities:
+        component_name = item["name"]
+        component_version = item.get("version")
+        for vulnerability in vulnerabilities:
+            vuln_id = vulnerability.get("id")
+            vuln_fix_versions = vulnerability.get("fix_versions")
+            vuln_description = vulnerability.get("description")
+
+            title = (
+                f"{vuln_id} in {component_name}:{component_version}"
+            )
+
+            description = ""
+            description += vuln_description
+
+            mitigation = None
+            if vuln_fix_versions:
+                mitigation = "Upgrade to version:"
+                if len(vuln_fix_versions) == 1:
+                    mitigation += f" {vuln_fix_versions[0]}"
+                else:
+                    for fix_version in vuln_fix_versions:
+                        mitigation += f"\n- {fix_version}"
+
+            finding = Finding(
+                test=test,
+                title=title,
+                cwe=1352,
```

Comment:
It looks like MITRE [prefers not to use this CWE](https://cwe.mitre.org/data/definitions/1352.html). [Searching through the Dojo codebase](https://github.com/search?q=repo%3ADefectDojo%2Fdjango-DefectDojo+1035&type=code), it appears that we tend to use [CWE-1035](https://cwe.mitre.org/data/definitions/1035.html), which MITRE also discourages.

We might want to start defaulting to [CWE-1395](https://cwe.mitre.org/data/definitions/1395.html) for these SCA issues, though I realize it would be somewhat disruptive to modify the default CWEs in all these different tools...

Reply:
I updated to default to 1395: https://github.com/DefectDojo/django-DefectDojo/pull/9696/commits/0820dc9491b18485da65601f20d10a900c13f8f9